### ISSUE TITLE
sigsev with self signed ca gitea base_url

### ISSUE THREAD

**Issue by poblin-orange** [Lerentis/terraform-provider-gitea#13]
**Describe the bug**
getting a sigsev issue

**To Reproduce**
using terraform plugin

**Expected behavior**

**Log Output**
```
 Error: Plugin did not respond                                                                                                                                                                           │

│                                                                                                                                                                                                         │

│   with provider["registry.terraform.io/lerentis/gitea"],                                                                                                                                                │

│   on gitea-configuration.tf line 5, in provider "gitea":                                                                                                                                                │

│    5: provider "gitea" {                                                                                                                                                                                │

│                                                                                                                                                                                                         │

│ The plugin encountered an error, and failed to respond to the                                                                                                                                           │

│ plugin.(*GRPCProvider).ConfigureProvider call. The plugin logs may contain                                                                                                                              │

│ more details.                                                                                                                                                                                           │

│                                                                                                                                                                                                         │

│ Stack trace from the terraform-provider-gitea_v0.11.0 plugin:                                                                                                                                           │

│                                                                                                                                                                                                         │

│ panic: runtime error: invalid memory address or nil pointer dereference                                                                                                                                 │

│ [signal SIGSEGV: segmentation violation code=0x1 addr=0x90 pc=0x6f73f6]                                                                                                                                 │

│                                                                                                                                                                                                         │

│ goroutine 27 [running]:                                                                                                                                                                                 │

│ code.gitea.io/sdk/gitea.(*Client).doRequest(0x0, {0xc50b79, 0x3}, {0xc51618, 0x5}, 0x0, {0x0?, 0x0})                                                                                                    │

│     code.gitea.io/sdk/gitea@v0.15.1/client.go:206 +0x76                                                                                                                                                 │

│ code.gitea.io/sdk/gitea.(*Client).getResponse(0xc00038f400?, {0xc50b79?, 0x0?}, {0xc51618?, 0xc0000a5318?}, 0x40d627?, {0x0?, 0x0?})                                                                    │

│     code.gitea.io/sdk/gitea@v0.15.1/client.go:289 +0x7d                                                                                                                                                 │

│ code.gitea.io/sdk/gitea.(*Client).getParsedResponse(0xc000498450?, {0xc50b79?, 0x6f6606?}, {0xc51618?, 0xc00007a2d0?}, 0xc0000a5301?, {0x0?, 0x0?}, {0xb17080, 0xc0003ba870})                           │

│     code.gitea.io/sdk/gitea@v0.15.1/client.go:311 +0x36                                                                                                                                                 │

│ code.gitea.io/sdk/gitea.(*Client).GetMyUserInfo(0xc000208390?)                                                                                                                                          │

│     code.gitea.io/sdk/gitea@v0.15.1/user.go:67 +0x65                                                                                                                                                    │

│ git.uploadfilter24.eu/terraform-provider-gitea/gitea.(*Config).Client(0xc0000a54c8)                                                                                                                     │

│     git.uploadfilter24.eu/terraform-provider-gitea/gitea/config.go:72 +0x3e7                                                                                                                            │

│ git.uploadfilter24.eu/terraform-provider-gitea/gitea.providerConfigure(0xc525d3?)                                                                                                                       │

│     git.uploadfilter24.eu/terraform-provider-gitea/gitea/provider.go:115 +0x27d                                                                                                                         │

│ github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Provider).Configure(0xc000078f00, {0xd79440, 0xc000498060}, 0xd?)                                                                          │

│     github.com/hashicorp/terraform-plugin-sdk/v2@v2.20.0/helper/schema/provider.go:287 +0x18f                                                                                                           │

│ github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ConfigureProvider(0xc0002a83a8, {0xd79440?, 0xc00047fb60?}, 0xc00021d8d8)                                              │

│     github.com/hashicorp/terraform-plugin-sdk/v2@v2.20.0/helper/schema/grpc_provider.go:557 +0x345                                                                                                      │

│ github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).Configure(0xc00019c8c0, {0xd79440?, 0xc00047f4a0?}, 0xc000418780)                                                                │

│     github.com/hashicorp/terraform-plugin-go@v0.12.0/tfprotov5/tf5server/server.go:556 +0x2ce                                                                                                           │

│ github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_Configure_Handler({0xc23c60?, 0xc00019c8c0}, {0xd79440, 0xc00047f4a0}, 0xc000489030, 0x0)                               │

│     github.com/hashicorp/terraform-plugin-go@v0.12.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:331 +0x170                                                                                       │

│ google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000003c0, {0xd7bec8, 0xc000378000}, 0xc0004339e0, 0xc00029dec0, 0x11ff758, 0x0)                                                                    │

│     google.golang.org/grpc@v1.48.0/server.go:1295 +0xb0b                                                                                                                                                │

│ google.golang.org/grpc.(*Server).handleStream(0xc0000003c0, {0xd7bec8, 0xc000378000}, 0xc0004339e0, 0x0)                                                                                                │

│     google.golang.org/grpc@v1.48.0/server.go:1636 +0xa1b                                                                                                                                                │

│ google.golang.org/grpc.(*Server).serveStreams.func1.2()                                                                                                                                                 │

│     google.golang.org/grpc@v1.48.0/server.go:932 +0x98                                                                                                                                                  │

│ created by google.golang.org/grpc.(*Server).serveStreams.func1                                                                                                                                          │

│     google.golang.org/grpc@v1.48.0/server.go:930 +0x28a                                                                                                                                                 │

│                                                                                                                                                                                                         │

│ Error: The terraform-provider-gitea_v0.11.0 plugin crashed!                                                                                                                                             │

│                                                                                                                                                                                                         │

│ This is always indicative of a bug within the plugin. It would be immensely                                                                                                                             │

└──
```
**Additional Data**
Important for reproducability.

- Terraform Version

│   "terraform_version": "1.3.1",                                                                                                                                                                         │

│   "platform": "linux_amd64",                                                                                                                                                                            │



- Operating System

from k8s/k3s, weave tf-controller

- Provider Version

0.11.0

**Comment by Lerentis** [comment]
Hi @poblin-orange

Thanks for reporting this (:

This seems like a nil pointer in the configuration when checking if the permission are okay. Can you share your provider configuration and a minimal terraform example to reproduce this? I did not see that since some versions

**Comment by poblin-orange** [comment]
hi @Lerentis 
Seems the issue is on the provider declaration:
```
variable "git_root_password" {}
variable "ops_domain" {}


provider "gitea" {
  # Configuration options
  base_url = format("https://git.%s", var.ops_domain)
  username = "root"
  password = var.git_root_password
  #insecure = true
  
}
```
The gitea instance is on a private corporate ca.
Uncommenting insecure = true just solved the issue

**Comment by Lerentis** [comment]
just to clarify: 

* the gitea instance is using a coporate (as in self signed) ca for the TLS endpoint
* The host from which terraform gets started does not trust the CA
* setting `insecure = false` (which is the default) fixes this segsev in your case?

**Comment by Lerentis** [comment]
Thanks for the context. as this is not blocking for you i will take a close look at this later this week and hopefully release a bugfix for that. i think i already know where this is coming from :/

**Comment by Lerentis** [comment]
this should hopefully be fixed with [v0.11.1  ](https://github.com/Lerentis/terraform-provider-gitea/releases/tag/v0.11.1)
